src/base/memory_monitor_sr.cpp

```diff
@@ -391,51 +478,76 @@ namespace cubperf
 } // namespace cubperf
 
 /* APIs */
+// TODO: API parameter check (after error code setting)
+using namespace cubperf;
+
 int mmon_initialize (const char *server_name)
 {
-  cubperf::mmon_Gl = new cubperf::memory_monitor (server_name);
+  int error = NO_ERROR;
+
+  assert (server_name != NULL);
 
-  if (cubperf::mmon_Gl == nullptr)
+  if (prm_get_bool_value (PRM_ID_MEMORY_MONITORING))
     {
-      /* error case */
-      return -1;
+      mmon_Gl = new memory_monitor (server_name);
+
+      if (mmon_Gl == nullptr)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (memory_monitor));
+	  error = ER_OUT_OF_VIRTUAL_MEMORY;
+	}
     }
+  return error;
+}
 
-  /* normal return */
-  return 0;
+void mmon_notify_server_start ()
+{
+  if (prm_get_bool_value (PRM_ID_MEMORY_MONITORING))
+    {
+      mmon_Gl->end_init_phase ();
+    }
 }
 
 void mmon_finalize ()
 {
-  delete cubperf::mmon_Gl;
+  if (prm_get_bool_value (PRM_ID_MEMORY_MONITORING))
+    {
+      delete mmon_Gl;
+    }
 }
 
-int mmon_add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, uint64_t size)
+void mmon_add_stat (THREAD_ENTRY *thread_p, MMON_STAT_ID stat_id, int64_t size)
 {
-  int error = NO_ERROR;
-
-  return error;
+  if (prm_get_bool_value (PRM_ID_MEMORY_MONITORING))
```

Comment:
```suggestion
  if (mmon_Gl != nullptr)
```
It's less coupled and faster only to use a module-related variable and to check the system parameter only in `mmon_initialize`.

Reply:
Done.